### ISSUE TITLE
bbi package installation failed

### ISSUE THREAD

**Issue by joybaker1** [metrumresearchgroup/bbr#689]
Recently I tried to use bbr::use_bbi() and package installation, and it gave me the following results. The package is from the your web and I tried other packages with similar error. I wonder where it is wrong?

![1](https://github.com/metrumresearchgroup/bbi/assets/169439419/1e874e8d-4dca-457b-b74f-5e422f6f1cdf)
![2](https://github.com/metrumresearchgroup/bbi/assets/169439419/be1896d7-fdfc-4e5b-9edf-6d11328febe9)

**Comment by joybaker1** [comment]
Hi Kyleam, thanks for the guide.
Waiting for the second solution to work, I extract the tarball trying to install, but it reminds me the message below that I am confused what to do.?
![514152000](https://github.com/metrumresearchgroup/bbr/assets/169439419/5019cf5f-e198-46b4-a138-8106403a20fd)

I am using bbr version 1.10.0.

**Comment by kyleam** [comment]
> I extract the tarball trying to install ...

If you've manually downloaded and extracted the bbi release tarball, try pointing the `bbr.bbi_exe_path` option in your R session to that location, something like

```R
options("bbr.bbi_exe_path" = "C:/Users/dell/Downloads/bbi_windows_amd64/bbi.exe")
```

And then check if `bbi_version()` works:

```R
bbr::bbi_version()
```

> I am using bbr version 1.10.0.

Hmm, so you have the `bbr::use_bbi()` fixes I pointed too.

Based on the output in your image, it looks like the call to bbi is failing.  So the `bbi_version()` call above may show the same failure.  If that doesn't show any more information, you could try invoking `bbi version` via processx directly in order to hopefully get an indication of what the underlying problem is:

```R
# Requires you have set "bbr.bbi_exe_path" option to where you downloaded `bbi.exe` to.
processx::run(getOption("bbr.bbi_exe_path"), "version")
```

**Comment by joybaker1** [comment]
Thanks for the guide, kyleam. I think it has been installed for now (though last line didn't give the path)

options("bbr.bbi_exe_path" = "C:/Program Files/bbi/bbi.exe")
bbi_version()
[1] "3.3.0"
Sys.which("bbi")
bbi 
 ""

**Comment by joybaker1** [comment]
I notice that I have to run the options() every time initiating R. Otherwise it is not initiated after bbr. Is it right?

> # check bbi version; make sure things are installed correctly
> bbi_version()
[1] ""
> options("bbr.bbi_exe_path" = "C:/Program Files/bbi/bbi.exe")
> # check bbi version; make sure things are installed correctly
> bbi_version()
[1] "3.3.0"

**Comment by kyleam** [comment]
Right, you need execute that every session, though you don't need to do it manually.  [Here](https://metrumresearchgroup.github.io/bbr/articles/getting-started.html#setting-bbi_exe_path) is the recommendation from the "Getting Started" vignette: 

> Please note that this must be set for every new R session, so it’s recommended to include the above snippet in your .Rprofile, replacing the "/data/apps/bbi" with the absolute path from your system/project, if you have installed it somewhere different.

**Comment by kyleam** [comment]
@joybaker1 If I understand correctly, you've got things up and running on your end.  It's still not clear to me why your initial `use_bbi` command failed, but, testing on our Windows instance, I'm unable to trigger an error.  So I'm going to close this issue, but please feel free to follow up with more details.  Thanks.

**Comment by joybaker1** [comment]
yes it is working by this alternative way. the use_bbi() gives different error message each time. It could be some widgets I missed.  I may work around it later.
Thanks very much.
